gpkit/constraints/sgp.py

```diff
@@ -110,6 +120,15 @@ def localsolve(self, solver=None, verbosity=1, x0=None, reltol=1e-4,
         self.gps = []  # NOTE: SIDE EFFECTS
         self.solver_outs = []
         self._results = []
+        if relax:
+            if verbosity > 0:
+                print("Using penalty CCP...")
+            relaxed_model = self.penalty_ccp()
```

Comment:
It seems a bit weird to me for the user-facing interface to be setting the `relax` kwarg of local solve. As we increase the number of choices for SP algorithms, we should decide whether they should all be accessed as different user-facing methods (`sp.penalty_ccp_solve(reltol=...)`), or whether they should be accessed via args and kwargs of `localsolve`. I would vote for the former. It allows cleaner algorithm-specific parameters.

Philosophically, I think it's nice for GPkit to expose users to the names (from the literature) of algorithms where possible. I think that builds trust (that we implemented things the same way as the literature) and encourages new algorithm developers to use GPkit to sandbox their algorithm (they get 'credit' because users actually interact with the name of their algorithm.

Reply:
Makes sense. Any thoughts @bqpd?

Reply:
Nvm, I agree with you and I'm just going to pull the trigger.